src/_pytest/fixtures.py

```diff
@@ -1245,21 +1217,57 @@ def __call__(self, function: FixtureFunction) -> FixtureFunction:
                 pytrace=False,
             )
 
-        # Type ignored because https://github.com/python/mypy/issues/2087.
-        function._pytestfixturefunction = self  # type: ignore[attr-defined]
-        return function
+        return fixture_definition
+
+
+# TODO: paramspec/return type annotation tracking and storing
+class FixtureFunctionDefinition:
+    def __init__(
```

Comment:
While this type is not public, it's seems like something that plugin might start to use directly. For this, we have an "opt in" mechanism which makes it clear the ctor is internal and can change between pytest versions without warning. The mechanism is to add a `_ispytest: bool = False,` parameter, always call with `_ispytest=True` inside of pytest, and issue a warning when it's `False`. There are many examples in the code you can copy from.